cl/citations/annotate_citations.py

```diff
@@ -0,0 +1,107 @@
+from typing import List
+
+from eyecite import annotate, clean_text
+from eyecite.models import CitationBase, NonopinionCitation
+
+from cl.search.models import Opinion
+
+
+def get_and_clean_opinion_text(opinion: Opinion) -> None:
+    """Memoize useful versions of an opinion's text as additional properties
+    on the Opinion object. This should be done before performing citation
+    extraction and annotation on an opinion.
+
+    :param opinion: The Opinion whose text should be parsed
+    """
+    if opinion.html_anon_2020:
+        opinion.source_text = opinion.html_anon_2020
+        opinion.cleaned_text = clean_text(
+            opinion.html_anon_2020, ["html", "all_whitespace"]
+        )
+        opinion.source_is_html = True
+    elif opinion.html_columbia:
+        opinion.source_text = opinion.html_columbia
+        opinion.cleaned_text = clean_text(
+            opinion.html_columbia, ["html", "all_whitespace"]
+        )
+        opinion.source_is_html = True
+    elif opinion.html_lawbox:
+        opinion.source_text = opinion.html_lawbox
+        opinion.cleaned_text = clean_text(
+            opinion.html_lawbox, ["html", "all_whitespace"]
+        )
+        opinion.source_is_html = True
+    elif opinion.html:
+        opinion.source_text = opinion.html
+        opinion.cleaned_text = clean_text(
+            opinion.html, ["html", "all_whitespace"]
+        )
+        opinion.source_is_html = True
+    elif opinion.plain_text:
+        opinion.source_text = opinion.plain_text
+        opinion.cleaned_text = clean_text(
+            opinion.plain_text, ["all_whitespace"]
+        )
+        opinion.source_is_html = False
```

Comment:
This could be simplified with something like:

```
for attr in ["html_anon_2020", "html_columbia", "html_lawbox", "html"]:
    text = getattr(opinion, attr)
    if text:
        opinion.source_text = text
        opinion.cleaned_text = clean_text(text, ["html", "all_whitespace"])
        opinion.source_is_html = True
        break
else:
    # Didn't hit the break; use plain text
    text = getattr(opinion, "plain_text")
    opinion.source_text = text
    opinion.cleaned_text = clean_text(text, ["all_whitespace"])
    opinion.source_is_html = False    
```